servers/lib/server/readclientversion.coffee

```diff
@@ -0,0 +1,43 @@
+KONFIG._CLIENTVERSION ?= KONFIG.version
+
+
+module.exports = readClientVersion = (callback = (->)) ->
+
+  fs   = require 'fs'
+  path = require 'path'
+
+  kallback = (version) ->
+    KONFIG._CLIENTVERSION = version
+    callback version
+
+  version = try
+    fs.readFileSync(
+      (path.join process.env.KONFIG_PROJECTROOT, './CLIENTVERSION'), 'utf-8'
+    ).trim()
+  catch
+    KONFIG.version
+
+
+  if KONFIG.environment in [ 'sandbox', 'production' ]
```

Comment:
env name approach is problematic where we might create problems for prod while we are on latest env.

scenario:
  * deploy to latest
  * switch version in latest (so it is prod)
  * a new prod machine spins up, gets the new version
  * we reverted the change on latest
  * we didnt run "switchClient" in prod machines.

i would suggest to use "KONFIG_EBENVNAME=koding-latest" property, it is set on compile time.